src/libraries/System.Private.CoreLib/src/System/IO/FileStream.Windows.cs

```diff
@@ -1249,19 +1249,23 @@ public override Task CopyToAsync(Stream destination, int bufferSize, Cancellatio
                 return base.CopyToAsync(destination, bufferSize, cancellationToken);
             }
 
-            StreamHelpers.ValidateCopyToArgs(this, destination, bufferSize);
-
-            // Bail early for cancellation if cancellation has been requested
-            if (cancellationToken.IsCancellationRequested)
-            {
-                return Task.FromCanceled<int>(cancellationToken);
-            }
+            ValidateCopyToArguments(destination, bufferSize);
 
             // Fail if the file was closed
             if (_fileHandle.IsClosed)
             {
                 throw Error.GetFileNotOpen();
             }
+            if (!CanRead)
```

Comment:
In which cases would a `FileStream` not be readable?

Reply:
If it's created with FileAccess.Write only.